### PR TITLE
Begin to factor DataBoxTag.hpp

### PR DESCRIPTION
**Pull request by kidder**
## Proposed changes

Starting to factor DataBoxTag.hpp into smaller parts.
This PR factors the empty structs that DataBox are inherited from, along with tag traits and `db::tag_name`.  More factoring will occur in a subsequent PR.

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [x] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### CODE REVIEW

**Comment by kidder** [comment]
clang-tidy error is boost root-finder

File: tests/Unit/DataStructures/DataBox/TestTags.hpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+#pragma once
+
+#include <string>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataBox/TagName.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace test {
```
Comment (nilsdeppe) [inline tests/Unit/DataStructures/DataBox/TestTags.hpp]: Why not `TestHelpers` for consistency? Then do a nested `namespace db`

File: tests/Unit/DataStructures/DataBox/Test_TagName.cpp [hunk 2]
```diff
@@ -0,0 +1,241 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <string>
+
+#include "DataStructures/DataBox/TagName.hpp"
```
Comment (nilsdeppe) [inline tests/Unit/DataStructures/DataBox/Test_TagName.cpp:9]: Also need `DataBox/Tag.hpp`

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Comment by kidder** [comment]
squashed in the fixups

**Review by nilsdeppe — DISMISSED** [review]
image

**Comment by kidder** [comment]
@fmahebert @moxcodes can one of you review this please?

**Review by fmahebert — APPROVED** [review]

**Comment by kidder** [comment]
wow fastest review ever 😛

**Comment by kidder** [comment]
@nilsdeppe I had to rebase

**Comment by fmahebert** [comment]
@kidder I'd been scrolling through it for a while already when you pinged 🙂

**Comment by kidder** [comment]
thanks @fmahebert !

**Review by nilsdeppe — APPROVED** [review]
@kidder you probably only rebased to get more seals ;)

image

**Comment by nilsdeppe** [comment]
Clang-tidy failure is boost root finder nonsense.

**Pull request merged**